### ISSUE TITLE
Example where softmax_cross_entropy_with_logits_v2 fails but softmax_cross_entropy_with_logits works

### ISSUE THREAD

**Issue by Rassibassi** [tensorflow/tensorflow#34336]
**System information**
- Have I written custom code: ?
- OS Platform and Distribution: Ubuntu 18.04
- TensorFlow installed from (source or binary): pip install
- TensorFlow version (use command below): v1.14.0-rc1-22-gaf24dc91b5 1.14.0
- Python version: Python 3.6.5 :: Anaconda, Inc.
- CUDA/cuDNN version: CPU
- GPU model and memory: CPU

**Describe the current behavior**
While implementing this paper [1] I found that using `softmax_cross_entropy_with_logits` leads to convergence and the correct results, whereas `softmax_cross_entropy_with_logits_v2` fails.
[1]: https://arxiv.org/abs/1906.07748

**Describe the expected behavior**
I suppose that `softmax_cross_entropy_with_logits_v2` should work just as `softmax_cross_entropy_with_logits`.

**Code to reproduce the issue**
Here a notebook:
https://github.com/Rassibassi/claude/blob/master/examples/tf_AutoEncoderForProbabilisticShapingAndAwgn.ipynb

In cell [8], just uncomment this line
```
# loss = tf.nn.softmax_cross_entropy_with_logits_v2(labels=s, logits=dec) # v2 does not work :(
```
and run for the code to fail, yet with `softmax_cross_entropy_with_logits` the result looks fine.

With `softmax_cross_entropy_with_logits`
```
epoch: 5 - outLossHat: -3.35 - always 1: 1.0
epoch: 10 - outLossHat: -4.08 - always 1: 1.0
epoch: 15 - outLossHat: -4.38 - always 1: 1.0
epoch: 20 - outLossHat: -4.47 - always 1: 1.0
epoch: 25 - outLossHat: -4.51 - always 1: 1.0
epoch: 30 - outLossHat: -4.51 - always 1: 1.0
epoch: 35 - outLossHat: -4.53 - always 1: 1.0
epoch: 40 - outLossHat: -4.53 - always 1: 1.0
epoch: 45 - outLossHat: -4.53 - always 1: 1.0
epoch: 50 - outLossHat: -4.53 - always 1: 1.0
```

with `softmax_cross_entropy_with_logits_v2`
```
epoch: 5 - outLossHat: -3.41 - always 1: 1.0
epoch: 10 - outLossHat: -3.7 - always 1: 1.0
epoch: 15 - outLossHat: -0.603 - always 1: 1.0
epoch: 20 - outLossHat: -0.00371 - always 1: 1.0
epoch: 25 - outLossHat: -0.0017 - always 1: 1.0
epoch: 30 - outLossHat: -0.00103 - always 1: 1.0
epoch: 35 - outLossHat: -0.000673 - always 1: 1.0
epoch: 40 - outLossHat: -0.00049 - always 1: 1.0
epoch: 45 - outLossHat: -0.000398 - always 1: 1.0
epoch: 50 - outLossHat: -0.000311 - always 1: 1.0
```

**Comment by gowthamkpr** [comment]
@Rassibassi Please go through the answer for this question [here](https://stats.stackexchange.com/questions/327348/how-is-softmax-cross-entropy-with-logits-different-from-softmax-cross-entropy-wi) and it explains the clear difference between `softmax_cross_entropy_with_logits ` and `softmax_cross_entropy_with_logits_v2` and this should give an intuition of why the error is occuring and let me know if it helps. Thanks!

**Comment by Rassibassi** [comment]
Thank you very much. And sorry, I thought this was an actual bug. I should have searched a bit more or read the doc's.
